### ISSUE TITLE
[OS X] Mouse didn't stay inside client

### ISSUE THREAD

**Issue by rained23** [ValveSoftware/Dota-2#212]
I have 1 external monitor that positioned as top and my built-in display as bottom.

Moving the mouse down will exit the mouse pointer from client and go to built-in display.

I think the mouse should stay and locked inside the client.

Maybe you can reproduce it by position an external display as top and built-in display as bottom.

My display arrangement
![image](https://cloud.githubusercontent.com/assets/907309/8509626/059ba354-22e0-11e5-8374-ad98c6ca47a0.png)

**Comment by gdrewb-valve** [comment]
If you look over #33 does that describe your situation?  Are you using fullscreen, windowed or borderless windows?  If you use fullscreen or a full borderless window (if you aren't already) does that work better?  Are you using the full monitor resolution for the monitor the game is on?

**Comment by rained23** [comment]
@gdrewb-valve Yes, I think is similar I am using full borderless window. I use "Use my monitor current resolution". I didnt use full screen cause it blacken my other display plus I cannot use mission control.

I think is similar with #33 maybe can merge this issue.

Update

I tried to use fullscreen, and this issue also happen in Fullscreen, its like the client also running on my built-in display thats why I can move my mouse between 2 display.

**Comment by budimanjojo** [comment]
Do you enable edge pan? If you enable it, play a bot game. After the game start the cursor should stay inside the window, the cursor don't stay inside the client is done by design which is what I like better than grabbing mouse when not in game

**Comment by rained23** [comment]
I have edge pan enabled.

How can you sure its by design ?

The mouse should stay in the client if you intend it to be there and does not bound to it if you focusing on
another  display, now the game is not playable when I move my mouse to bottom as it will go passthrough the client and enter my builtin screen and I might accidentally clicked on something on my bottom display.

I think the best way the mouse should unbound itself from the client when you change your display focus, 
example by switching to Desktop or other fullscreen apps.

**Comment by budimanjojo** [comment]
Have you tried playing a bot game?
Here's what happen in my case:
1. Mouse can go outside the client when you are not in a game. 
2. When the game starts, the mouse is grabbed in the Dota 2 Reborn. 
3. When spectating a game with other views except "Free Camera", the mouse can go wherever you want it to be. 
4. When spectating a game with "Free Camera" view, mouse will be grabbed again because you control the cursor in game. 
This is what I see in my case, so I think this is the design of the new Dota 2 Reborn which is a better approach than the old mouse grabbing all the time which is causing other problems like alt tabbing etc

**Comment by rained23** [comment]
Not happen on me for no 2. Try reproduce my issue.

Play any game, custom or anything, move mouse faster to the bottom.

Result : the mouse will exit Reborn client and go to your desktop and careful dragging your file into the trash hehe.

The currently working left and right only, bottom should be locked too.

Btw you need to have same arrangement display as me, and run Reborn on Top.

**Comment by budimanjojo** [comment]
What is your monitor resolution? I think you are using borderless window with the monitor resolution? Try changing it to a smaller resolution or "Use monitor resolution" instead. Btw I'm a Linux user but I think it should be the same.

**Comment by rained23** [comment]
Oh maybe its only happen to OS X. If your mouse still bounded when playing a game  moving mouse to the bottom, this mean only happen to OS X. I use my monitor resolution, dont think changing to a smaller resolution is a good fix for me plus my mouse dpi is high and sensitive .

**Comment by budimanjojo** [comment]
No, it also happens to me if I use borderless window on full resolution, because dota 2 will launch below the top panel. So the bottom will be below your monitor. Can you post a screenshot at how your Dota 2 window displayed? You have to try using fullscreen if you want full resolution and use borderless window only if you want smaller resolution

**Comment by rained23** [comment]
I use borderless window on full resolution. Using fullscreen will disable all my display, my external screen will be Reborn client, and other display will be black. I cannot use mission control or anything.

**Comment by budimanjojo** [comment]
I don't understand what do you mean by disable all your display. Then I think you can try using launch option to "shorten" your dota height. What is your desktop resolution? For example if you are using 1920x1080 display, try putting this into your dota launch option:
-windowed -noborder -h 1060 -w 1920
Now I think bottom edge panning should be working.

**Comment by davidw-valve** [comment]
In Dota2 Source1 do you experience this problem?

**Comment by rained23** [comment]
@davidw-valve No, my mouse bound to the client for Source 1 with the same arrangement. Its only happen in Source 2

**Comment by rained23** [comment]
I just tested the new update, it fix this issue. Thanks.

Update

After some games, the problems happen again, my mouse will unbound from the client and go straight to my second monitor at bottom.

**Comment by gdrewb-valve** [comment]
Is this still happening?  I saw something from the larger team that a fix had been made in this area.

**Comment by gdrewb-valve** [comment]
Closing pending confirmation that this is still active.

**Comment by benlilley** [comment]
Sorry to comment on an old thread but I'm still having this problem. Nothing I can do seems to stop the mouse leaving the dota window and leaving the monitor while playing. I have screens side by side and when I edge pan on the right the mouse ends up on my other desktop and not in the dota window.

I've tried this on all window settings (fullscreen, exclusive fullscreen and windowed) but they all have the same problem.

OS X 10.11.2

EDIT: I take this back, it seems to be resolved since last time I tried. Previously I had the same behaviour in the menu screens vs in the game, whilst when I just tried in a match now the behaviour seems to be as I expected so I think this has been resolved.
